corehq/ex-submodules/phonelog/utils.py

```diff
@@ -277,3 +280,26 @@ def force_close_subreport(self):
                 device_model=log.get('device_model'),
             ) for log in logs
         ).encode('utf-8'))
+
+
+def clear_device_log_request(domain, xform):
+    from corehq.apps.ota.models import DeviceLogRequest
+    user_subreport = _get_logs(xform.form_data, 'user_subreport', 'user')
+    username = (user_subreport[0].get('username') if user_subreport
+                else cached_user_id_to_username(xform.user_id))
+    device_id = xform.form_data.get('device_id')
+    try:
+        if not (username and device_id):
+            raise DeviceLogRequest.DoesNotExist()
+        log_request = DeviceLogRequest.objects.get(
+            domain=domain,
+            username=username,
+            device_id=device_id,
+        )
+    except DeviceLogRequest.DoesNotExist:
+        msg = "Forced log submission, but no corresponding request found."
```

Comment:
@esoergel  Wanted to check if in cases when `force_logs` is set though HQ can't find any corresponding Device Log Request, do we still accept logs associated with that request ?

Reply:
Yes, though it logs that as an error so hopefully we can trace down what's happening.  If this did occur, the user would essentially gets stuck receiving `force_logs` in their heartbeat requests until we noticed and manually deleted the log request.  Not the worst thing to happen, but figured it'd be wise to put some preemptive debugging info in there.

Reply:
Cool. I just wanted to see the possibility of if we can use this `force_logs` flag just to force logs anytime even if it's not because of a DeviceLogRequest entry for ex - If user manually wants to submit the logs by going to a menu.  I will let you know if we indeed decide to make that change on mobile in case you wanna remove this error logging in that case.